.ci/kata-doc-to-script.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+[ -n "$DEBUG" ] && set -x
+
+script_name="${0##*/}"
+
+typeset -r warning="WARNING: Do *NOT* run the generated script without reviewing it carefully first!"
+
+# github markdown markers used to surround a code block. All text within the
+# markers is rendered in a fixed font.
+typeset -r block_open="\`\`\`bash"
+typeset -r block_close="\`\`\`"
+
+# convention used in all documentation to represent a non-privileged users
+# shell prompt. All lines starting with this value inside a code block are
+# commands the user should run.
+typeset -r code_prompt="\$ "
+
+# files are expected to match this regular expression
+typeset -r extension_regex="\.md$"
+
+strict="no"
+require_commands="no"
+check_only="no"
+verbose="no"
+
+usage()
+{
+	cat <<EOT
+Usage: ${script_name} [options] <markdown-file> [<script-file>]
+
+This script will convert a github-flavoured markdown document file into a
+bash(1) script to stdout by extracting the bash code blocks.
+
+
+Options:
+
+  -c : check the file but don't create the script (sets exit code).
+  -h : show this usage.
+  -r : require atleast one command block to be found.
+  -s : strict mode - perform extra checks.
+  -v : verbose mode.
+
+${warning}
+
+Example usage:
+
+  $ ${script_name} foo.md foo.md.sh
+
+${warning}
+
+EOT
+
+	exit 0
+}
+
+die()
+{
+	local msg="$*"
+
+	echo "ERROR: $msg" >&2
+	exit 1
+}
+
+script_header()
+{
+	cat <<-EOT
+	#!/bin/bash
+	#----------------------------------------------
+	# WARNING: Script auto-generated from '$file'.
+	#
```

Comment:
I wonder if we should inject a license header in here as well (I don't feel there is a big issue if we do or do not, but having one makes the generated code ownership/license situation clear....)